core/src/main/scala/cats/effect/TestConsole.scala

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2018 Typelevel
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cats.effect
+import cats.Show
+import cats.effect.concurrent.Ref
+import cats.syntax.show._
+
+class TestConsole(out1: Ref[IO, List[String]],
+                  out2: Ref[IO, List[String]],
+                  out3: Ref[IO, List[String]])
+    extends Console[IO] {
+  override def putStrLn[A: Show](a: A): IO[Unit] =
+    out1.update(acc => acc ::: a.show :: Nil)
+  override def putStr[A: Show](a: A): IO[Unit] =
+    out2.update(acc => acc ::: a.show :: Nil)
+  override def putError[A: Show](a: A): IO[Unit] =
+    out3.update(acc => acc ::: a.show :: Nil)
+  override val readLn: IO[String] = IO.pure("test")
```

Comment:
This should probably be configurable with a sequence of predefined inputs, repeating (or throwing an exception) in case the input ends